src/Sulu/Bundle/MediaBundle/Controller/MediaController.php

```diff
@@ -203,16 +203,29 @@ public function cgetAction(Request $request)
         $listBuilder = $this->getListBuilder($request, $fieldDescriptors, $types);
         $listBuilder->setParameter('locale', $locale);
         $listResponse = $listBuilder->execute();
-        $count = $listBuilder->count();
+
+        $previewImageIds = [];
+        foreach ($listResponse as $listItem) {
+            $previewImageId = $listItem['previewImageId'] ?? null;
+            if ($previewImageId) {
+                $previewImageIds[] = $previewImageId;
+            }
+        }
+        $previewImageFormats = $this->mediaManager->getFormatUrls($previewImageIds, $locale);
```

Comment:
This loads the entire preview image right? The thing is that the preview image is already displayed for me, isn't it for you? If it is already working I don't see why we should make all these db requests.

Reply:
yes, the `MediaManager` probably load the whole media for this. the alternative would be to load the `name`, `version`, `subversion` and `mimeType` of the preview image in the list response. but that would complicate the media list xml a lot.

Reply:
@danrot it display the image because the imageAction only checks for the `id` but the image is then always recalculated on the server as the filename and folder does not match where it is stored.

Instead using the mediaManager we could join the columns of the previewImage we need:

```php
                    $listResponse[$i]['previewImageId'],
                    $listResponse[$i]['previewImageName'],
                    $listResponse[$i]['previewImageVersion'],
                    $listResponse[$i]['previewImageSubversion'],
                    $listResponse[$i]['previewImageMimeType']
```

Reply:
Would IMO also make the code easier to read. Can you maybe also make a very short benchmark if it is really improving the performance compared to what is currently done in this PR?

Reply:
regarding to that the image is already working for you. looks like this depends on the system for `pdf` files. i adapted the pr description to use an `xls` file. does that work for you too?

Reply:
Nope, does not seem to work for `xls` files

Reply:
That make sense as the false mimeType was given (original media instead of preview media) and pdf is supported and xls is not to generate images.

Reply:
i adjusted the pr to use the `ListBuilder` for loading the required properties of the preview image. this makes the media list config more complicated, but we dont need any additional request.

did not do any benchmark because i am not sure how to do that in a useful manner on my machine. but if anybody wants to do that, he can reset the last commit to compare the two strategies 🙂